.github/workflows/prepare-for-minor-release.yml

```
@@ -28,19 +38,31 @@ jobs:
         id: define-version-variables
         run: |
           delimiter=.
-          version_parts=($(echo ${{ inputs.release_branch_name }} | tr $delimiter '\n'))
-          version_parts[1]=$((version_parts[1]+1))
-          NEXT_MINOR_RELEASE=$(IFS=$delimiter ; echo "${version_parts[*]}" | sed 's/\(.*\)\.x/\1/')
-          NEXT_RELEASE_VERSION_CONSTRAINT="~${NEXT_MINOR_RELEASE}"
-          echo "NEXT_MINOR_RELEASE=${NEXT_MINOR_RELEASE}" >> $GITHUB_OUTPUT
+          version_parts=($(echo "${{ inputs.release_branch_name }}" | tr $delimiter '\n'))
+
+          if [ "${{ inputs.release_type }}" = "major" ]; then
+            version_parts[0]=$((version_parts[0]+1))
+            version_parts[1]=0
+          elif [ "${{ inputs.release_type }}" = "minor" ]; then
+            version_parts[1]=$((version_parts[1]+1))
+          else
+            echo "Invalid release_type: ${{ inputs.release_type }}"
+            exit 1
+          fi
+
+          NEXT_RELEASE=$(IFS=$delimiter ; echo "${version_parts[*]}" | sed 's/\\(.*\\)\\.x/\\1/')
+          NEXT_RELEASE_VERSION_CONSTRAINT="~${NEXT_RELEASE}"
+          THIS_RELEASE_VERSION_CONSTRAINT=$(echo "${{ inputs.release_branch_name }}" | sed 's/\\(.*\\)\\.x/~\\1/')
+          MAIN_BRANCH_ALIAS="${NEXT_RELEASE}.x-dev"
+
+          echo "NEXT_MINOR_RELEASE=${NEXT_RELEASE}" >> $GITHUB_OUTPUT
```

Should this echo statement be updated, and should the `next_minor_release` output be updated on line 32 above?

```suggestion
          echo "NEXT_RELEASE=${NEXT_RELEASE}" >> $GITHUB_OUTPUT
```

---

I’m having trouble understanding the purpose of the NEXT_RELEASE variable — specifically, what it should be set to and how it should be used when creating a major release. Should it always represent a bump to the next minor version, even when we’re preparing a major release branch (like 3.x)? Or should its behavior differ depending on the release type?